debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -302,15 +302,30 @@ public ReplicationStream startStreaming(Lsn offset, WalPositionLocator walPositi
             LOGGER.debug("starting streaming from LSN '{}'", lsn);
         }
 
-        try {
-            return createReplicationStream(lsn, walPosition);
-        }
-        catch (Exception e) {
-            String message = "Failed to start replication stream at " + lsn;
-            if (e.getMessage().matches(".*replication slot .* is active.*")) {
-                message += "; when setting up multiple connectors for the same database host, please make sure to use a distinct replication slot name for each.";
+        int maxRetries = originalConfig.getInteger(PostgresConnectorConfig.MAX_RETRIES);
+        int tryCount = 0;
+        while (true) {
+            try {
+                return createReplicationStream(lsn, walPosition);
+            }
+            catch (Exception e) {
+                String message = "Failed to start replication stream at " + lsn;
+                if (++tryCount > maxRetries) {
+                    if (e.getMessage().matches(".*replication slot .* is active.*")) {
+                        message += "; when setting up multiple connectors for the same database host, please make sure to use a distinct replication slot name for each.";
+                    }
+                    throw new DebeziumException(message, e);
+                }
+                else {
+                    LOGGER.warn(message);
+                    LOGGER.warn("Waiting 2s and retrying, attempt number " + tryCount + " over " + maxRetries);
```

Comment:
Could you please log it as a single messgae? Also please use formatted maessage with `{}` placeholders instead of building the string.

Reply:
Yes, will take this into account